src/main/java/uk/ac/ebi/atlas/download/FileDownloadController.java

```diff
@@ -144,4 +155,60 @@ public FileDownloadController(ExperimentFileLocationService experimentFileLocati
             zipOutputStream.close();
         }
     }
+
+    @GetMapping(value = "json/experiments/download/zip/check",
+            produces = "application/json;charset=UTF-8")
+    @ResponseStatus(HttpStatus.OK)
+    @ResponseBody
+    public String
+    checkMultipleExperimentsFileValid(@RequestParam(value = "accession", defaultValue = "") List<String> accessions)
+    {
+        List<Experiment> experiments = new ArrayList<>();
+
+        accessions.stream().forEach(accession -> {
+            try{
+                experiments.add(experimentTrader.getPublicExperiment(accession));
+            } catch (Exception e) {
+                LOGGER.debug("Invalid experiment accession: {}", accession);
+            }
+        });
+
+        ImmutableList<ExperimentFileType> fileTypeCheckList = ImmutableList.of(
+                ExperimentFileType.QUANTIFICATION_RAW,
+                ExperimentFileType.NORMALISED,
+                ExperimentFileType.EXPERIMENT_DESIGN);
+
+        var filePaths = experiments.stream()
+                .map(experiment ->
+                        Map.entry(experiment.getAccession(),
+                                fileTypeCheckList.stream()
+                                        .map(fileType -> fileType == ExperimentFileType.EXPERIMENT_DESIGN ?
+                                                ImmutableList.of(experimentFileLocationService
+                                                        .getFilePath(experiment.getAccession(), fileType)) :
+                                                experimentFileLocationService
+                                                        .getFilePathsForArchive(experiment.getAccession(), fileType))
+                                        .collect(ImmutableList.toImmutableList())
+
+                        ))
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+
+        var invalidFilesList = filePaths.keySet().stream()
```

Comment:
No need to iterate over the keys and get the value for each key, you can just do that with `filePaths.entrySet().stream()`.

Reply:
Right. I reformated it using `entrySet()`.